src/helpers.ts

```diff
@@ -90,6 +90,7 @@ export function createDefaultClient(
 }
 
 export function createDefaultEthSignClient(
+  ethAddress: string,
```

Comment:
Can't you get the address from the `ethers` signer that's passed in?

Reply:
Yep it can be done like that, however that function should be a `async` function

Reply:
like 
```js
export async function createDefaultEthSignClientAsync(
  dappchainKey: string,
  dappchainEndpoint: string,
  chainId: string,
  wallet: ethers.Signer
): Promise<{ client: Client; publicKey: Uint8Array; address: Address }> {
  const defaultClientObj = createDefaultClient(dappchainKey, dappchainEndpoint, chainId)
  const ethAddress = await wallet.getAddress()

  defaultClientObj.client.txMiddleware = [
    new NonceTxMiddleware(
      new Address('eth', LocalAddress.fromHexString(ethAddress)),
      defaultClientObj.client
    ),
    new SignedEthTxMiddleware(wallet)
  ]

  return { ...defaultClientObj }
}
```

Reply:
It's a new function so making it `async` is fine by me.